src/browser/api_protocol/api_handlers/system.ts

```diff
@@ -115,6 +116,30 @@ function didFail(e: any): boolean {
     return e !== undefined && e.constructor === Error;
 }
 
+const dosURL = 'https://openfin.co/documentation/desktop-owner-settings/';
+
+async function getServiceConfiguration(identity: Identity, message: APIMessage) {
+    const { name } = message.payload;
+    const response = await System.getServiceConfiguration();
+
+    if (didFail(response)) {
+        throw response;
+    }
+
+    if (!Array.isArray(response)) {
+        throw new Error(`Settings in desktop owner settings are not configured correctly, please see
+         ${dosURL} for configuration information`);
+    }
+
+    const config = response.find(service => service.name === name);
```

Comment:
Any reason we are doing filtering at the `handler` level versus the `api` level?

Reply:
I like throwing safely from there, want to keep the system call as safe as possible just in case we use it internally

Reply:
Is this just for OF services?  I wonder if we could make this a more generic "ask RVM" API so that services (& other apps?) can take new RVM features without runtime upgrades.  RVM is making a big push to have everyone on latest, so if we make a generic API (and maybe not document it?) then RVM can expose some new features without runtime changes.  Just a thought, maybe not for this PR.

Reply:
Yes, in principle I like this. And because its whats under the covers we can swap this on top of something like that in the future.